src/platform/plugins/shared/controls/public/controls/esql_control/types.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+import type { PublishesPanelTitle, PublishingSubject } from '@kbn/presentation-publishing';
+import type { ESQLControlState as ControlState } from '@kbn/esql/public';
+import type { DefaultControlState } from '../../../common';
+import type { DefaultControlApi } from '../types';
+import type { CanClearVariables } from '../../types';
+
+interface PublishesSelectedOptions {
```

Comment:
Why is `PublishesSelectedOptions` broken into a separate type? I don't see it imported anywhere else or used anywhere else. How about deleting `PublishesSelectedOptions` type?

Reply:
Ofc https://github.com/elastic/kibana/pull/202875/commits/0611ecf30299910457467360e3ae0a12d6c93c7e